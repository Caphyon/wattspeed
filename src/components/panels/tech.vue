<template>
  <div class="mainContent">
    <template v-if="requestLimit">
      <div class="quota-exceeded">
        <h3 class="text--center text--danger text--strong">
          You have reached your request quota per hour, try again later.
        </h3>
        <svg class="watch" width="200" height="200" viewBox="0 0 185 185" fill="none" xmlns="http://www.w3.org/2000/svg">
          <path d="M92.5 185C143.586 185 185 143.586 185 92.5C185 41.4137 143.586 0 92.5 0C41.4137 0 0 41.4137 0 92.5C0 143.586 41.4137 185 92.5 185Z" fill="url(#paint0_linear)"/>
          <path d="M174.4 92.2998C174.4 137.3 137.9 173.8 92.8999 173.8C47.8999 173.8 11.3999 137.3 11.3999 92.2998C11.3999 47.2998 47.8999 10.7998 92.8999 10.7998C137.9 10.7998 174.4 47.1998 174.4 92.2998Z" stroke="url(#paint2_linear)" stroke-width="2" fill="none"/>
          <path d="M92.8999 161.8C131.284 161.8 162.4 130.684 162.4 92.2998C162.4 53.916 131.284 22.7998 92.8999 22.7998C54.5161 22.7998 23.3999 53.916 23.3999 92.2998C23.3999 130.684 54.5161 161.8 92.8999 161.8Z" stroke="url(#paint1_linear)" stroke-width="2" fill="none"/>
          <path opacity="0.5" d="M92.6001 92.7002V42.7002" stroke="white" stroke-width="2" stroke-linecap="round" class="line"/>
          <path opacity="0.5" d="M112.6 113.3L92.6001 92.7998" stroke="white" stroke-width="2" stroke-linecap="round" class="line line--alt"/>
          <path d="M92.5 97.5C95.2614 97.5 97.5 95.2614 97.5 92.5C97.5 89.7386 95.2614 87.5 92.5 87.5C89.7386 87.5 87.5 89.7386 87.5 92.5C87.5 95.2614 89.7386 97.5 92.5 97.5Z" fill="white"/>
          <defs>
          <linearGradient id="paint0_linear" x1="92.5" y1="0" x2="92.5" y2="322.5" gradientUnits="userSpaceOnUse">
          <stop stop-opacity="0"/>
          <stop offset="1"/>
          </linearGradient>
          <linearGradient id="paint1_linear" x1="92.55" y1="162.749" x2="93.05" y2="54.2491" gradientUnits="userSpaceOnUse">
          <stop stop-color="#fff"/>
          <stop offset="1" stop-color="#fff" stop-opacity="0"/>
          </linearGradient>
          <linearGradient id="paint2_linear" x1="93.255" y1="9.7507" x2="92.67" y2="136.719" gradientUnits="userSpaceOnUse">
          <stop stop-color="#FF8282"/>
          <stop offset="1" stop-color="#FF8282" stop-opacity="0"/>
          </linearGradient>
          </defs>
      </svg>
      </div>
    </template>
    <div v-else class="sections">
        <tech-section @tooManyRequests="handleTooManyRequests"></tech-section>
        <html5-section @tooManyRequests="handleTooManyRequests"></html5-section>
        <performance-section @tooManyRequests="handleTooManyRequests"></performance-section>
        <security-section @tooManyRequests="handleTooManyRequests"></security-section>
        <mixed-section @tooManyRequests="handleTooManyRequests"></mixed-section>
        <accesibility-section @tooManyRequests="handleTooManyRequests"></accesibility-section>
    </div>
  </div>
</template>

<script>
import Vue from "vue";
import BaseSection from "./../sections/base-section.vue";
import Tech from "./../sections/tech-section.vue";
import Html5 from "./../sections/html5-section.vue";
import Performance from "./../sections/performance-section.vue";
import Security from "./../sections/security-section.vue";
import Mixed from "./../sections/mixed-section.vue";
import Accesibility from "./../sections/accesibility-section.vue";

Vue.component("base-section", BaseSection);
Vue.component("tech-section", Tech);
Vue.component("html5-section", Html5);
Vue.component("performance-section", Performance);
Vue.component("security-section", Security);
Vue.component("mixed-section", Mixed);
Vue.component("accesibility-section", Accesibility);

export default {
  data: function(){
    return {
      requestLimit: false
    }
  },
  computed: {
    tab() {
      return this.$parent.tab;
    }
  },
  methods: {
    changePanel(tab) {
      this.$parent.changePanel(tab);
    },
    handleTooManyRequests(){
      this.requestLimit = true;
    }
  }
};
</script>
