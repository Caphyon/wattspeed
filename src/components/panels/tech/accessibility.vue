<template>
    <container>
      <template slot="header">
        <accessibility-status :score="score" :issues="issues" :warnings="warnings" v-if="score != 100"></accessibility-status>
        <h3 v-else class="text--success text--center text--strong mb0">
          No accessibility issues found.
        </h3>
      </template>

      <div v-if="!data" class="state">
        <svg viewBox="0 0 386.9 345.3" xmlns="http://www.w3.org/2000/svg" width="250">
          <path d="m121.7 215.3 54.5 54.5 120-120c14.1 19.7 22.5 43.9 22.5 70 0 66.6-53.9 120.5-120.5 120.5s-120.5-54-120.5-120.5 54-120.5 120.5-120.5c25.6 0 49.4 8 68.9 21.6" fill="none" stroke="#6ed499" stroke-linecap="round" stroke-linejoin="round" stroke-miterlimit="10" stroke-width="10" class="state--success"/>
          <g class="state--sparks" style="transform-origin: center center">
            <path d="m157 23.5c-1.6 4.7-3 6.9-6.4 10.4-1.1-.9-1.7-1.4-2.8-2.3 2.3-3.5 3.1-5.5 3.6-9.5 2.2.5 3.3.8 5.6 1.4zm-71 79c-2.9 0-5.2 2.3-5.2 5.2s2.3 5.2 5.2 5.2 5.2-2.3 5.2-5.2-2.3-5.2-5.2-5.2zm155.4-60.5c0-2.1-1.7-3.7-3.7-3.7s-3.7 1.7-3.7 3.7c0 2.1 1.7 3.7 3.7 3.7s3.7-1.6 3.7-3.7zm113.5 92.9c-.4-3.7-3.7-6.3-7.3-5.9-3.7.4-6.3 3.7-5.9 7.3.4 3.7 3.7 6.3 7.3 5.9 3.7-.4 6.3-3.7 5.9-7.3z" fill="#ffd600"/>
            <path d="m336.8 103.3c-7.4 2.6-10.9 4.8-16.6 10.4-2.1-1.8-3.2-2.6-5.3-4.4 5.3-7.8 8.9-11.1 17.2-16 1.9 4 2.8 6 4.7 10zm-146.7-31.3c0-2.1-1.7-3.7-3.7-3.7-2.1 0-3.7 1.7-3.7 3.7s1.7 3.7 3.7 3.7 3.7-1.6 3.7-3.7zm-143.1-6.3c-.7-4-1.4-5.9-3.7-9.1-1 .6-1.5.9-2.5 1.4 1.3 3.1 1.7 4.8 1.6 8.1 1.9-.2 2.8-.3 4.6-.4zm4 85.9c-.2-1.8-1.8-3.1-3.7-2.9s-3.1 1.8-2.9 3.7c.2 1.8 1.8 3.1 3.7 2.9s3.1-1.8 2.9-3.7zm-30.9 91.4c10.8-.7 16.2-2.3 25.5-7.5-1.3-2.9-1.9-4.3-3.2-7.1-8.6 2.8-13.2 3.3-22.1 2.2 0 4.9-.1 7.4-.2 12.4z" fill="#f7931e"/>
            <path d="m205.3 0c1.1 7.8 2.6 11.6 7 18.3-2.1 1.8-3.2 2.6-5.3 4.4-6.6-6.7-9.2-10.9-12.3-20 4.2-1.1 6.3-1.6 10.6-2.7zm-95.5 2.6c0-1.3-1.1-2.4-2.4-2.4s-2.4 1.1-2.4 2.4 1 2.4 2.4 2.4 2.4-1 2.4-2.4zm184.2 21.9c0-1.3-1.1-2.4-2.4-2.4s-2.4 1.1-2.4 2.4 1.1 2.4 2.4 2.4c1.4 0 2.4-1.1 2.4-2.4zm-268.5 87.3c-6 3.5-8.6 5.9-12.4 11.5 1.5 1.3 2.3 1.9 3.9 3.2 4.1-4.1 6.6-5.6 12-7.5-1.5-2.9-2.2-4.3-3.5-7.2zm361.4 70.7c-9.4-5.3-14.8-6.9-25.5-7.7-.5 3.1-.7 4.6-1.2 7.7 8.7 2.4 12.9 4.5 19.6 10.3 2.9-4.1 4.3-6.2 7.1-10.3z" fill="#f93556"/>
            <path d="m110.5 74.3c-8.7-3.7-12.5-7-17.7-15 3.6-1.9 5.5-2.9 9.1-4.8 2.8 6.9 5.3 10 11.6 14.2-1.2 2.2-1.8 3.4-3 5.6zm178.8-11.6c-3.4 3.6-5.5 5-10.2 6.9.5 1.7.7 2.5 1.2 4.2 5.8-1.1 8.7-2.3 13.6-5.8-1.9-2.1-2.8-3.2-4.6-5.3zm-108.4-49.2c0-1.3-1.1-2.4-2.4-2.4s-2.4 1.1-2.4 2.4 1.1 2.4 2.4 2.4 2.4-1.1 2.4-2.4zm204 89.3c-.2-2.1-2.1-3.5-4.1-3.3-2.1.2-3.5 2.1-3.3 4.1s2.1 3.5 4.1 3.3 3.5-2 3.3-4.1zm-384.6 85.7c-.8 2.2.3 4.6 2.5 5.4s4.6-.3 5.4-2.5-.3-4.6-2.5-5.4-4.6.3-5.4 2.5z" fill="#7ddafb"/>
          </g>
        </svg>
      </div>
      <accessibility-item v-for="item in filteredData" :data="item" :key="item.title"></accessibility-item>
    <template slot="filters">
          <filter-item :filterCategories="this.initialFilterTypes"
                        @onFilter="updateFilteredResults">
          </filter-item>
        </template>
    </container>
</template>
<script>
import Vue from "vue";
import Container from "./container";
import Item from "./items/accessibility-item";
import FilterItem from './items/filter-item';

let Constant = require('./../../../assets/utils/consts.js')

Vue.component("container", Container);
Vue.component("accessibility-item", Item);
Vue.component("filter-item", FilterItem);

import AccessibilitySection from "./../../sections/accesibility-section.vue";

export default {
  mixins: [AccessibilitySection],
  data() {
    return {
      initialData: [],
      filteredData: [],
      filtered_results: [Constant.WARNINGS, Constant.ERRORS],
      initialFilterTypes: [],
      desc:
        "The power of the Web is in its universality. Access by everyone regardless of disability is an essential aspect. (Tim Berners-Lee, W3C Director and inventor of the World Wide Web)"
    };
  },
    watch: {
    data: function() {
      if (this.data) {
        this.initialData = this.filteredData = this.computeInitialData(this.data);
        this.initialFilterTypes = this.computeFilterCategories(this.data);
      }
    },
  },
  methods: {
    computeInitialData(data) {
      return data.map(crtItem=> {
        let newItem ={};
        newItem = crtItem

        if(crtItem.severity == 'critical' || crtItem.severity == "serious") {
          newItem.class = 'alert--danger';
          newItem.type = Constant.ERRORS;
        } else if(crtItem.severity == 'moderate' || crtItem.severity == "minor") {
          newItem.class = 'alert--warning';
          newItem.type = Constant.WARNINGS;
        }

        return newItem;
      });
    },
    computeFilterCategories(initialData) {
      return [... new Set(initialData.map(item => item.type))];
    },
    updateFilteredResults(value) {
      this.filteredData = this.initialData.filter(crtItem => {
        return value.includes(crtItem.type);
      });
    }
  },
};
</script>
