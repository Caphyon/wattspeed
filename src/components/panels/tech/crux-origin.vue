<template>
  <pair-container class="lh-vars">
    <div class="lh-audit-group--metrics"
         :class="data.origin ? 'lh-audit-group mb0' : ''">
        <div class="mb2">
          <div class="flex pt-2">
            <div>
              <img
                :alt="`Core web vitals ${getCWVStatus('origin')}`"
                :src="getCWVStatusSvg('origin')"
                height="60"
                class="mx-auto block"
                style="max-height: 60px;"/>
            </div>
            <p v-if="data.origin" class="my-auto">
              <span v-html="getGroupText('origin')"/>
            </p>
            <p v-else class="my-auto">
              <span class="lh-audit-group__title">Origin summary</span>
              <span v-html="getNoDataText('origin')"/>
            </p>
          </div>
        </div>
        <div class="lh-metrics-container">
          <template v-for="(metricId, index) in Object.keys(sections)">
            <div
              v-if="data.origin
                && data.origin.record
                && data.origin.record.metrics[sections[metricId].id]"
              class="lh-metric lh-metric--numeric border-0"
              :class="`lh-metric--${getStatusClass('origin', sections[metricId].id)}`"
              :key="index">
              <div class="lh-metric__innerwrap_old gap-y-1">
                <div class="lh-metric__title">
                  {{ sections[metricId].title }}
                  <span v-if="sections[metricId].coreVitals" title="Core Web Vitals"> ðŸ“ˆ </span>
                </div>
                <div class="lh-metric__value_old">
                  {{ getPercentile('origin', sections[metricId].id) }}
                </div>
                <div class="progress crux_progress col-span-2 col-start-2 h-auto">
                  <div
                    class="progress-bar crux_progress--fast font-bold rounded-lg"
                    :class="{ '-mr-3 pr-4' : fastHasNext('origin', sections[metricId].id) }"
                    role="progressbar"
                    v-if="getData('origin', sections[metricId].id, 0, true)"
                    :style="'flex-grow:' + getData('origin', sections[metricId].id, 0)"
                    :aria-valuenow="getData('origin', sections[metricId].id, 0)"
                    aria-valuemin="0"
                    aria-valuemax="100">
                    {{ getData('origin', sections[metricId].id, 0, true) }}%
                  </div>
                  <div
                    class="progress-bar crux_progress--avg font-bold rounded-lg"
                    :class="{ '-mr-3 pr-4' : avgHasNext('origin', sections[metricId].id) }"
                    role="progressbar"
                    v-if="getData('origin', sections[metricId].id, 1, true)"
                    :style="'flex-grow:' + getData('origin', sections[metricId].id, 1)"
                    :aria-valuenow="getData('origin', sections[metricId].id, 1)"
                    aria-valuemin="0"
                    aria-valuemax="100">
                    {{ getData('origin', sections[metricId].id, 1, true) }}%
                  </div>
                  <div
                    class="progress-bar crux_progress--slow rounded-lg font-bold"
                    role="progressbar"
                    v-if="getData('origin', sections[metricId].id, 2, true)"
                    :style="'flex-grow:' + getData('origin', sections[metricId].id, 2)"
                    :aria-valuenow="getData('origin', sections[metricId].id, 2)"
                    aria-valuemin="0"
                    aria-valuemax="100">
                    {{ getData('origin', sections[metricId].id, 2, true) }}%
                  </div>
                </div>
              </div>
            </div>
            <div
              v-else
              class="lh-metric lh-metric--numeric border-0"
              :class="`lh-metric--notapplicable`"
              :key="index">
              <div class="lh-metric__innerwrap_old gap-y-1">
                <div class="lh-metric__title">
                  {{ sections[metricId].title }}
                  <span v-if="sections[metricId].coreVitals" title="Core Web Vitals"> ðŸ“ˆ </span>
                </div>
                <div class="lh-metric__value_old">N/A</div>
                <div class="progress crux_progress col-span-2 col-start-2 h-auto">
                  <div
                    class="progress-bar crux_progress--notapplicable rounded-lg text-sm font-bold"
                    role="progressbar"
                    :style="'flex-grow: 100'"
                    :aria-valuenow="100"
                    aria-valuemin="0"
                    aria-valuemax="100">
                    N/A
                  </div>
                </div>
              </div>
            </div>
          </template>
        </div>
  </div>
  </pair-container>
</template>
<script>
import Vue from "vue";
import PairContainer from "./pair-container.vue";

Vue.component("pair-container", PairContainer);

import CrUXSection from "../../sections/crux-section.vue";

export default {
  mixins: [CrUXSection],
  data() {
    return {
      panelName: "crux-origin",
      activeFirst: false,
      activeSecond: true,
    };
  },
};
</script>

<style lang="scss" scoped>
.lh-vars {
  --color-not-applicable: var(--color-gray-600);
  --default-padding: 8px;
  --audit-group-margin-bottom: calc(var(--default-padding) * 6);
  --report-line-height: 24px;
  --report-border-color-secondary: var(--color-blue-border);
  --color-blue-border: #374060;
  --score-icon-size: 12px;
  --score-icon-margin-left: 6px;
  --score-icon-margin-right: 14px;
  --score-icon-margin: 0 var(--score-icon-margin-right) 0 var(--score-icon-margin-left);
  --audit-description-padding-left: calc(var(--score-icon-size) + var(--score-icon-margin-left) + var(--score-icon-margin-right));
  --color-gray-100: #F5F5F5;
  --color-gray-200: #E0E0E0;
  --color-gray-600: #757575;
  --color-gray-700: #616161;
  --color-gray-800: #424242;
  --color-bg-container: #272e38;
  --color-red: rgba(206, 55, 77);
  --color-orange: rgba(225, 167, 48);
  --color-green: rgba(57, 176, 126);

  color: rgb(203, 213, 225);
}

ul {
  li:not(:last-of-type) {
    margin-bottom: 1rem;
  }
}

.lh-audit-group {
  margin-bottom: var(--audit-group-margin-bottom);
  position: relative;
}

.lh-audit-group--metrics {
  margin-bottom: calc(var(--audit-group-margin-bottom) / 2);
}

.lh-audit-group__title {
  text-transform: uppercase;
  font-weight: 500;
}

.lh-metrics-container {
  display: grid;
  grid-auto-rows: 1fr;
  grid-template-columns: 1fr 1fr;
  grid-column-gap: var(--report-line-height);
  margin-bottom: var(--default-padding);
}

.lh-metric {
  border-top: 1px solid var(--report-border-color-secondary);
}

.lh-table-column--numeric {
  text-align: right;
  word-break: normal;
}

.lh-metric__innerwrap_old {
  display: grid;
  grid-template-columns: var(--audit-description-padding-left) 10fr 3fr;
  align-items: center;
  padding: 10px 0;
}

.lh-metric__value_old {
  white-space: nowrap; /* No wrapping between metric value and the icon */
  font-weight: 500;
  justify-self: end;
}

/* Progress */
.crux_progress--fast {
  background-color: var(--color-green);
}

.crux_progress--avg {
  background-color: var(--color-orange);
}

.crux_progress--slow {
  background-color: var(--color-red);
}

.crux_progress--notapplicable {
  background-color: var(--color-gray-800);
  color: var(--color-gray-700) !important;
}

.progress {
  display: flex;
  overflow: hidden;
  border-radius: 4px;
  background-color: var(--color-bg-container);

  &-bar {
    height: 17px!important;
    font-size: 0.75rem;
    display: flex;
    flex-direction: column;
    justify-content: center;
    white-space: nowrap;
    text-align: center;
    line-height: 1rem;
    color: var(--color-gray-100);
  }
}
.flex {
  display: flex;
}
.-mr-3 {
  margin-right: -0.75rem!important;
}
.rounded-lg {
  border-radius: 0.5rem;
}
.pr-4 {
  padding-right: 1rem!important;
}
.pt-2 {
  padding-right: .5rem!important;
}
.w-full {
  width: 100%;
}
.text-sm {
  font-size: 0.875rem;
}

.font-bold {
  font-weight: 700;
}

.mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.my-auto {
  margin-top: auto;
  margin-bottom: auto;
}

.block {
  display: block;
}

.col-span-2 {
  grid-column: span 2 / span 2;
}

.col-start-2 {
  grid-column-start: 2;
}

.h-auto {
  height: auto;
}

.gap-y-1 {
  row-gap: 0.25rem!important;
}

.p-1 {
  padding-top: .25rem;
  padding-bottom: .25rem;
}

.lh-metric .lh-metric__innerwrap:before,
.lh-metric .lh-metric__innerwrap_old:before,
.lh-scorescale-range:before {
  content: "";
  width: var(--score-icon-size);
  height: var(--score-icon-size);
  display: inline-block;
  margin: var(--score-icon-margin);
  margin-top: 2px;
}

.lh-metric--pass .lh-metric__innerwrap::before,
.lh-metric--pass .lh-metric__innerwrap_old::before {
  border-radius: 100%;
  background: var(--color-green);
}

.lh-metric--pass .lh-metric__value,
.lh-metric--pass .lh-metric__value_old {
  color: var(--color-green);
}

.lh-metric--average .lh-metric__innerwrap::before,
.lh-metric--average .lh-metric__innerwrap_old::before{
  background: var(--color-orange);
}

.lh-metric--average .lh-metric__value,
.lh-metric--average .lh-metric__value_old {
  color: var(--color-orange);
}

.lh-metric--fail .lh-metric__innerwrap::before,
.lh-metric--error .lh-metric__innerwrap::before,
.lh-metric--fail .lh-metric__innerwrap_old::before,
.lh-metric--error .lh-metric__innerwrap_old::before {
  border-left: calc(var(--score-icon-size) / 2) solid transparent;
  border-right: calc(var(--score-icon-size) / 2) solid transparent;
  border-bottom: var(--score-icon-size) solid var(--color-red);
  height: 0px;
  width: 0px;
}

.lh-metric--fail .lh-metric__value,
.lh-metric--fail .lh-metric__value_old {
  color: var(--color-red);
}

.lh-metric--notapplicable .lh-metric__innerwrap::before,
.lh-metric--notapplicable .lh-metric__innerwrap_old::before {
  border-radius: 100%;
  background: var(--color-not-applicable);
}
.lh-metric--notapplicable .lh-metric__value,
.lh-metric--notapplicable .lh-metric__value_old {
  color: var(--color-not-applicable);
}
</style>
