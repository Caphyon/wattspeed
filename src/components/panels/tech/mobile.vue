<template>
    <pair-container>
        <template v-if="!error" slot="header">
          <h3 v-if="mobile_friendly" class="text--success text--center text--strong mb0">
            {{mobile_friendly}}
          </h3>
          <h3 v-if="not_mobile_friendly" class="text--danger text--center text--strong mb0">
            {{not_mobile_friendly}}
          </h3>
        </template>
        <div class="scrollable__container">
          <div>
            <mobile-item v-for="item in items" :data="item" :key="item.title"></mobile-item>
          </div>
          <div class="mobile">
            <img class="mobile--screen" v-if="screenshot" :src="screenshot"/>
            <img class="mobile--frame" src="/assets/icons/mobile-frame.svg" height="305"/>
          </div>
        </div>
    </pair-container>
</template>
<script>
import Vue from "vue";
import MobileSection from "./../../sections/mobile-section";
import PairContainer from "./pair-container";
import Item from "./items/mobile-item";

import { marked } from 'marked';

Vue.component("pair-container", PairContainer);
Vue.component("mobile-item", Item);

export default {
  mixins: [MobileSection],
  data() {
    return {
      desc:
        "The Google's Mobile-Friendly test will analyze a URL and report if the page has a mobile-friendly design."
    };
  },
  computed: {
    screenshot() {
      if (this.loading || !this.data || !this.data.audits["final-screenshot"].details) {
        return "";
      }
      return this.data.audits["final-screenshot"].details.data;
    },
    items() {
      const items = [];

      if (this.loading || !this.data.audits) {
        return {};
      }

      for (let key in this.data.audits) {
        if (key === "viewport" || key === "font-size" || key === "plugins" || key === "content-width") {
          const value = this.data.audits[key];
          items.push({
            passed: value.score !== null ? value.score : 0,
            title: marked.parseInline(value.title),
          });
        }
      }

      return items;
    },
  }
};
</script>
<style lang="scss" scoped>
.mobile {
  position: relative;
  margin: auto;

  &--frame {
    transform: translateZ(0);
  }

  &::before {
    content: "";
    position: absolute;
    background-image: url(data:image/svg+xml;base64,PHN2ZyBlbmFibGUtYmFja2dyb3VuZD0ibmV3IDAgMCA1NTQuNSA3MTIuNSIgdmlld0JveD0iMCAwIDU1NC41IDcxMi41IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHhtbG5zOnhsaW5rPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5L3hsaW5rIj48bGluZWFyR3JhZGllbnQgaWQ9ImEiPjxzdG9wIG9mZnNldD0iMCIgc3RvcC1jb2xvcj0iI2ZmZiIgc3RvcC1vcGFjaXR5PSIwIi8+PHN0b3Agb2Zmc2V0PSIuNSIgc3RvcC1jb2xvcj0iI2ZmZiIvPjxzdG9wIG9mZnNldD0iMSIgc3RvcC1jb2xvcj0iI2ZmZiIgc3RvcC1vcGFjaXR5PSIwIi8+PC9saW5lYXJHcmFkaWVudD48bGluZWFyR3JhZGllbnQgaWQ9ImIiIGdyYWRpZW50VW5pdHM9InVzZXJTcGFjZU9uVXNlIiB4MT0iODkuNSIgeDI9Ijg5LjUiIHhsaW5rOmhyZWY9IiNhIiB5MT0iNzIwIiB5Mj0iMy41ODUzIi8+PGxpbmVhckdyYWRpZW50IGlkPSJjIiBncmFkaWVudFVuaXRzPSJ1c2VyU3BhY2VPblVzZSIgeDE9IjQ1OS41IiB4Mj0iNDU5LjUiIHhsaW5rOmhyZWY9IiNhIiB5MT0iNzE0IiB5Mj0iLS4zNTMyIi8+PGxpbmVhckdyYWRpZW50IGlkPSJkIiBncmFkaWVudFVuaXRzPSJ1c2VyU3BhY2VPblVzZSIgeDE9IjAiIHgyPSI1NTUuMTEwMiIgeGxpbms6aHJlZj0iI2EiIHkxPSI5MC43NSIgeTI9IjkwLjc1Ii8+PGxpbmVhckdyYWRpZW50IGlkPSJlIiBncmFkaWVudFVuaXRzPSJ1c2VyU3BhY2VPblVzZSIgeDE9IjAiIHgyPSI1NTUuMTEwMiIgeGxpbms6aHJlZj0iI2EiIHkxPSIyNjUuMjUiIHkyPSIyNjUuMjUiLz48bGluZWFyR3JhZGllbnQgaWQ9ImYiIGdyYWRpZW50VW5pdHM9InVzZXJTcGFjZU9uVXNlIiB4MT0iMCIgeDI9IjU1NS4xMTAyIiB4bGluazpocmVmPSIjYSIgeTE9IjQzOS43NSIgeTI9IjQzOS43NSIvPjxsaW5lYXJHcmFkaWVudCBpZD0iZyIgZ3JhZGllbnRVbml0cz0idXNlclNwYWNlT25Vc2UiIHgxPSIwIiB4Mj0iNTU0LjUiIHhsaW5rOmhyZWY9IiNhIiB5MT0iNjE0LjI1IiB5Mj0iNjE0LjI1Ii8+PHBhdGggZD0ibTg4IDBoM3Y3MTIuNWgtM3oiIGZpbGw9InVybCgjYikiLz48cGF0aCBkPSJtNDU4IDBoM3Y3MTIuNWgtM3oiIGZpbGw9InVybCgjYykiLz48cGF0aCBkPSJtMCA4OS4zaDU1NC41djNoLTU1NC41eiIgZmlsbD0idXJsKCNkKSIvPjxwYXRoIGQ9Im0wIDI2My44aDU1NC41djNoLTU1NC41eiIgZmlsbD0idXJsKCNlKSIvPjxwYXRoIGQ9Im0wIDQzOC4zaDU1NC41djNoLTU1NC41eiIgZmlsbD0idXJsKCNmKSIvPjxwYXRoIGQ9Im0wIDYxMi44aDU1NC41djNoLTU1NC41eiIgZmlsbD0idXJsKCNnKSIvPjwvc3ZnPg==);
    width: 250px;
    background-position: center;
    background-size: cover;
    height: 100%;
    top: 0;
    left: 50%;
    transform: translateX(-50%);
    opacity: .3;
  }

  &--screen {
    position: absolute;
    top: calc(50% - 2.5px);
    left: 50%;
    transform: translateY(-50%) translateX(-50%);
    width: 90%;
    z-index: 1;
    border-radius: 18px;
  }
}
</style>
