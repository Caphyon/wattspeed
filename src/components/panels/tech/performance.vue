<template>
    <pair-container>
      <template slot="header">
        <div v-if="!error" class="section__header--stats px2 py1">
          <performance-score :score="mobileScore" type="Mobile"></performance-score>
          <performance-score :score="desktopScore" type="Desktop"></performance-score>
        </div>
      </template>
      <div class="scrollable__container">
        <div>
          <h4 class="text--center text--muted mt0 mb1">Mobile</h4>
         <performance-item v-for="(msg,index) in filteredMobileData" :data="msg" :key="index"></performance-item>
        </div>
        <div>
          <h4 class="text--center text--muted mt0 mb1">Desktop</h4>
          <performance-item v-for="(msg, index) in filteredDesktopData" :data="msg" :key="index"></performance-item>
        </div>
      </div>
      <template slot="filters">
        <filter-item :filterCategories="this.initialFilterTypes"
                     @onFilter="updateFilteredResults">
        </filter-item>
      </template>
    </pair-container>
</template>

<script>

import Vue from "vue";
import PerformanceSection from "../../sections/performance-section";
import PairContainer from "./pair-container";
import Item from "./items/performance-item";
import FilterItem from "./items/filter-item";

let Constant = require('./../../../assets/utils/consts.js')

Vue.component("pair-container", PairContainer);
Vue.component("performance-item", Item);
Vue.component("filter-item", FilterItem);

export default {
  mixins: [PerformanceSection],

  data() {
    return {
      initialDesktopData: [],
      initialMobileData: [],
      initialFilterTypes: [],
      filteredMobileData: [],
      filteredDesktopData: [],
      desc:
        "Performance measures the performance of a page for mobile devices and desktop devices. It fetches the url twice, once with a mobile user-agent, and once with a desktop-user agent."
    };
  },

  watch: {
    essentialData: function() {
      if (this.essentialData[Constant.DESKTOP] && this.essentialData[Constant.MOBILE]) {
        this.initialDesktopData = this.filteredDesktopData = this.computeInitialData(this.essentialData[Constant.DESKTOP]);
        this.initialMobileData = this.filteredMobileData = this.computeInitialData(this.essentialData[Constant.MOBILE]);
        this.initialFilterTypes = this.computeFilterCategories(this.initialMobileData, this.initialDesktopData);
      }
    },
  },

  methods: {
    computeInitialData(data) {
      return data.map(crtItem=> {
        let newItem ={};
        newItem.title = crtItem.title;
        newItem.level = crtItem.score;
        newItem.msg = crtItem.description;

        if (crtItem.score >= 0.9) {
          newItem.type = Constant.SUCCESS;
        } else if (crtItem.score >= 0.65) {
          newItem.type = Constant.WARNINGS;
        } else  {
          newItem.type = Constant.ERRORS;
        }

        return newItem;
      });
    },

    computeFilterCategories(initialMobileData, initialDesktopData) {
      let a =  [...new Set(initialMobileData.map(item => item.type))];
      let b =  [...new Set(initialDesktopData.map(item => item.type))];

      return [...new Set(a.concat(b))];
    },
    mobileMessages() {
      return this.extractMessages("mobile");
    },
    desktopMessages() {
      return this.extractMessages("desktop");
    },

    updateFilteredResults(value) {
      this.filteredMobileData = this.initialMobileData.filter(crtItem => {
        return value.includes(crtItem.type);
      });
      this.filteredDesktopData = this.initialDesktopData.filter(crtItem => {
        return value.includes(crtItem.type);
      });

    }
  }
};
</script>
