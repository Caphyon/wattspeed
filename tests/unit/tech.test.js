import { shallowMount } from "@vue/test-utils";
import tech from "../../src/components/panels/tech.vue"; // name of your Vue component
import tech_section from "../../src/components/sections/tech-section.vue";
import html5_section from "../../src/components/sections/html5-section.vue";
import speed_section from "../../src/components/sections/performance-section.vue";
import security_section from "../../src/components/sections/security-section.vue";
import mixed_section from "../../src/components/sections/mixed-section.vue";
import accessibility_section from "../../src/components/sections/accesibility-section.vue";

describe("Testing tech.vue component", () => {
  it("Should render corectly", () => {
    const wrapper = shallowMount(tech);

    // controll if the component is instance
    expect(wrapper.isVueInstance()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();

    // check if the divs are presents
    expect(wrapper.find('.mainContent').exists()).toBe(true);
    expect(wrapper.find('.sections').exists()).toBe(true);

    // check if all the components are imported correctly
    expect(wrapper.contains(tech_section)).toBe(true);
    expect(wrapper.contains(html5_section)).toBe(true);
    expect(wrapper.contains(speed_section)).toBe(true);
    expect(wrapper.contains(security_section)).toBe(true);
    expect(wrapper.contains(mixed_section)).toBe(true);
    expect(wrapper.contains(accessibility_section)).toBe(true);
  });

});
