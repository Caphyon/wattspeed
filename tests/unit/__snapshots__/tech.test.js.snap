// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing tech.vue component Should render corectly 1`] = `
<div
  class="mainContent"
>
  <div
    class="sections"
  >
    <tech-section-stub />
     
    <html5-section-stub />
     
    <performance-section-stub />
     
    <security-section-stub />
     
    <mixed-section-stub />
     
    <accesibility-section-stub />
  </div>
</div>
`;
